src/modules/graphics/wrap_GraphicsShader.lua

```diff
@@ -324,9 +324,16 @@ local function createShaderStageCode(stage, code, lang, gles, glsl1on3, gammacor
 		GLSL.FUNCTIONS,
 		GLSL[stage].FUNCTIONS,
 		custom and GLSL[stage].MAIN_CUSTOM or GLSL[stage].MAIN,
-		((lang == "glsl1" or glsl1on3) and not gles) and "#line 0" or "#line 1",
-		code,
 	}
+	if defines then
+		for k, v in pairs(defines) do
+			if type(k) == "string" and (type(v) == "string" or type(v) == "number" or type(v) == "boolean") then
+				table.insert(lines, string.format("#define %s %s", k, tostring(v)))
```

Comment:
I think true/false should be converted to 1 and 0, since you can't do `#if true` in GLSL to my knowledge (whereas `#if 1` and `#if 0` work).